### ISSUE TITLE
Scatter plot filltozeroy fails on multi-signed values

### ISSUE THREAD

**Issue by MattAmos** [plotly/plotly.py#2322]
Apologies if this is a duplicate issue, but a similar one could not be found. 

For scatter plot data containing both positive and negative values, `fill="tozeroy"` does not correctly fill under the trace as I expected. I'm not sure if this is a future feature or current bug, but I couldn't find documentation which states that scatter plot filling for both negative and positive values was not supported. 
```python
plotly.__version__
```
4.5.4

 ```python
from plotly import graph_objects as go
import numpy as np

x = np.linspace(0, 100)
y = np.random.rand(100) - 0.5

fig = go.Figure(data=go.Scattergl(x=x, y=y, fill="tozeroy"))
fig.show()
```
![image](https://user-images.githubusercontent.com/15097165/77732035-2b0e1e80-7058-11ea-981c-92519497c152.png)

**Comment by marcus-maro** [comment]
@MattAmos did you find a solution to this problem? I am experiencing the same issue.

**Comment by MattAmos** [comment]
> @MattAmos did you find a solution to this problem? I am experiencing the same issue.

Hi @marcus-maro, I didn't find a solution using `go.Scattergl`, but did find a workaround using `go.Scatter`.

```python
from plotly import graph_objects as go
import numpy as np

x = np.linspace(0, 100)
y = np.random.rand(100) - 0.5

fig = go.Figure(data=go.Scattergl(x=x, y=y, fill="tozeroy"))
fig.show()

fig = go.Figure(data=go.Scatter(x=x, y=y, fill="tozeroy"))
fig.show()
```

![image](https://user-images.githubusercontent.com/15097165/93301364-af896f00-f83b-11ea-8b31-4827f8812f51.png)
![image](https://user-images.githubusercontent.com/15097165/93301384-b617e680-f83b-11ea-86b1-37b52ec4ed49.png)

I'll reopen this issue as it seems like this is a genuine issue with `go.Scattergl` not performing in the same manner as `go.Scatter`.

**Comment by nicolaskruchten** [comment]
Indeed, `scattergl` does not have as good support for filled areas as `scatter` and this is a bug, although it'll have to be fixed in Plotly.js... there are a few current known shortcomings with this combination of features: https://github.com/plotly/plotly.js/issues/2291 and https://github.com/plotly/plotly.js/issues/4017
